modules/nf-core/raxmlng/main.nf

```diff
@@ -3,32 +3,52 @@ process RAXMLNG {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/raxml-ng:1.0.3--h32fcf60_0' :
-        'biocontainers/raxml-ng:1.0.3--h32fcf60_0' }"
+        'https://depot.galaxyproject.org/singularity/raxml-ng:1.2.2--h6747034_1' :
+        'biocontainers/raxml-ng:1.2.2--h6747034_1' }"
 
     input:
-    path alignment
+    tuple val(meta), path(alignment), val(model)
 
     output:
-    path "*.raxml.bestTree", emit: phylogeny
-    path "*.raxml.support" , optional:true, emit: phylogeny_bootstrapped
-    path "versions.yml"    , emit: versions
+    // either bestTree or bootstraps file is created, depending on options given
+    tuple val(meta), path("*.raxml.bestTree")  , emit: phylogeny             , optional:true
+    tuple val(meta), path("*.raxml.bootstraps"), emit: phylogeny_bootstrapped, optional:true
+    path "versions.yml"                        , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    // fix random seed for reproducibility if not specified in command line
+    if (!(args ==~ /.*--seed.*/)) {args += " --seed=42"}
     """
     raxml-ng \\
         $args \\
         --msa $alignment \\
+        --model $model \\
         --threads $task.cpus \\
-        --prefix output
+        --prefix ${prefix}
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
         raxmlng: \$(echo \$(raxml-ng --version 2>&1) | sed 's/^.*RAxML-NG v. //; s/released.*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def touch_support = args.contains('--bootstrap') || args.contains('--bs-trees') ? "touch ${prefix}.raxml.bootstraps" : ""
+    """
+    # Create stub output files
+    touch ${prefix}.raxml.bestTree
```

Comment:
If it's optional is there a way to know when `.bestTree` files are generated from the arguments ?

Reply:
Hmm, I think that it is either bestTree or the bootstrap file (from the other tests), so could possibly make the stub either/or. But I'm not certain.

Reply:
I've swapped to make the file dependent on the bootstrap or not.